server/channels/api4/channel_test.go

```diff
@@ -144,6 +144,37 @@ func TestUpdateChannel(t *testing.T) {
 	client := th.Client
 	team := th.BasicTeam
 
+	t.Run("update channel with another channel's name", func(t *testing.T) {
```

Comment:
We should actually write a store layer test for this. That way it's easier to assert the actual error type and the field. The client layer test asserts the error string which is brittle and can change.